src/Engines/MeilisearchEngine.php

```diff
@@ -378,7 +373,28 @@ public function deleteIndex($name)
      */
     public function deleteAllIndexes()
     {
-        return $this->meilisearch->deleteAllIndexes();
+        $tasks = [];
+        $offset = 0;
+        $limit = 20;
+        $hasMorePages = true;
+
+        while ($hasMorePages) {
+            $query = new IndexesQuery();
+            $query->setOffset($offset)->setLimit($limit);
+
+            $indexes = $this->meilisearch->getIndexes($query);
+            foreach ($indexes->getResults() as $index) {
+                $tasks[] = $index->delete();
+            }
+
+            if ($indexes->count() < $limit) {
+                $hasMorePages = false;
+            }
```

Comment:
if number of search exactly match a multiple of $limit, this implementation would perform an unnecessary extra request.

This can be prevented with following additional check :

```suggestion
            if ($indexes->count() < $limit || count($tasks) === $indexes->getTotal()) {
                $hasMorePages = false;
            }
```